velox/functions/prestosql/tests/ArrayNGramTest.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <optional>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class ArrayNGramTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testNgram(
+      const std::vector<std::vector<std::optional<T>>>& inputArray,
```

Comment:
nit: since this method test one row at a time, you can pass `const std::vector<std::optional<T>> inputArray` here and wrap it later when calling makeNullableArrayVector for brevity.